src/config/logtest-config.c

```diff
@@ -0,0 +1,114 @@
+/* Copyright (C) 2015-2020, Wazuh Inc.
+ * Copyright (C) 2009 Trend Micro Inc.
+ * All right reserved.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation
+ */
+
+#include "logtest-config.h"
+
+
+const char *enabled = "enabled";
+const char *threads = "threads";
+const char *max_sessions = "max_sessions";
+const char *session_timeout = "session_timeout";
+
+
+int Read_Logtest(XML_NODE node) {
+
+    for (int i = 0; node[i]; i++) {
+
+        if (!node[i]->element) {
+            merror(XML_ELEMNULL);
+            return OS_INVALID;
+        }
+
+        else if (!node[i]->content) {
+            merror(XML_VALUENULL, node[i]->element);
+            return OS_INVALID;
+        }
+
+        else if (!strcmp(node[i]->element, enabled)) {
+            if (strcmp(node[i]->content, "yes") && strcmp(node[i]->content, "no")) {
+                mwarn(XML_VALUEERR, node[i]->element, node[i]->content);
+                return OS_INVALID;
+            }
+            strcpy(w_logtest_conf.enabled, node[i]->content);
+        }
+
+        else if (!strcmp(node[i]->element, threads)) {
+            if (!strcmp(node[i]->content, "auto")) {
+                w_logtest_conf.threads = get_nproc();
+                continue;
+            }
+
+            char *end;
+            long value = strtol(node[i]->content, &end, 10);
+
+            if (value < 0 || value > 65534 || *end) {
+                mwarn(XML_VALUEERR, node[i]->element, node[i]->content);
+                return OS_INVALID;
+            } else if (value > LOGTEST_LIMIT_THREAD) {
+                mdebug2(LOGTEST_INV_NUM_THREADS, LOGTEST_LIMIT_THREAD);
+                w_logtest_conf.threads = LOGTEST_LIMIT_THREAD;
+            } else {
+                w_logtest_conf.threads = (unsigned short) value;
+            }
+        }
+
+        else if (!strcmp(node[i]->element, max_sessions)) {
+            char *end;
+            long value = strtol(node[i]->content, &end, 10);
+
+            if (value < 0 || value > 65534 || *end) {
+                mwarn(XML_VALUEERR, node[i]->element, node[i]->content);
+                return OS_INVALID;
+            } else if (value > LOGTEST_LIMIT_MAX_SESSIONS) {
+                mdebug2(LOGTEST_INV_NUM_USERS, LOGTEST_LIMIT_MAX_SESSIONS);
```

Comment:
We should warn the user about this as we are applying a different value from which the user gave.